plantdescriptionengine/src/main/java/eu/arrowhead/core/plantdescriptionengine/PdeMain.java

```diff
@@ -0,0 +1,381 @@
+import eu.arrowhead.core.plantdescriptionengine.consumedservices.serviceregistry.dto.SrSystem;
+import eu.arrowhead.core.plantdescriptionengine.pdtracker.PlantDescriptionTracker;
+import eu.arrowhead.core.plantdescriptionengine.pdtracker.backingstore.FilePdStore;
+import eu.arrowhead.core.plantdescriptionengine.pdtracker.backingstore.PdStore;
+import eu.arrowhead.core.plantdescriptionengine.providedservices.pde_mgmt.PdeManagementService;
+import eu.arrowhead.core.plantdescriptionengine.providedservices.pde_monitor.PdeMonitorService;
+import eu.arrowhead.core.plantdescriptionengine.providedservices.pde_monitorable.PdeMonitorableService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import se.arkalix.ArServiceRecordCache;
+import se.arkalix.ArSystem;
+import se.arkalix.core.plugin.HttpJsonCloudPlugin;
+import se.arkalix.core.plugin.or.OrchestrationOption;
+import se.arkalix.core.plugin.or.OrchestrationPattern;
+import se.arkalix.core.plugin.or.OrchestrationStrategy;
+import se.arkalix.net.http.client.HttpClient;
+import se.arkalix.security.identity.OwnedIdentity;
+import se.arkalix.security.identity.TrustStore;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.InetSocketAddress;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Properties;
+
+public final class PdeMain {
+
+    private static final Logger logger = LoggerFactory.getLogger(PdeMain.class);
+
+    private static final String PDE_SYSTEM_NAME = "plantdescriptionengine";
+    private static final String APP_PROPS_FILENAME = "application.properties";
+
+    /**
+     * Helper method for retrieving required properties. If the specified
+     * property is not present, an {@code IllegalArgumentException} is thrown.
+     *
+     * @param appProps A set of properties.
+     * @param propName Name of the property to retrieve.
+     * @return The property with the given name.
+     */
+    private static String getProp(final Properties appProps, final String propName) {
+        final String result = appProps.getProperty(propName);
+        if (result == null) {
+            throw new IllegalArgumentException("Missing field '" + propName + "' in application properties.");
+        }
+        return result;
+    }
+
+    /**
+     * @param appProps Configurations used for this instance of the Plant
+     *                 Description Engine.
+     * @return HTTP client useful for consuming Arrowhead services.
+     */
+    static HttpClient createHttpClient(final Properties appProps) {
+
+        final boolean secureMode = Boolean.parseBoolean(getProp(appProps, "server.ssl.enabled"));
```

Comment:
Generally, it is a good practice to move string literals to constants. The only exceptions are the descriptions, log and error messages.

Reply:
Fixed